### ISSUE TITLE
[java] PrimitiveWrapperInstantiation: no violation on 'new Boolean(val)'

### ISSUE THREAD

**Issue by dykov** [pmd/pmd#3595]
**Affects PMD Version:** 6.40.0

**Rule:** [PrimitiveWrapperInstantiation ](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#primitivewrapperinstantiation)

**Description:** No violation on Boolean object instantiation.
The same code has violation for the [BooleanInstantiation](https://pmd.github.io/latest/pmd_rules_java_performance.html#booleaninstantiation) rule (PMD ver. 6.36.0).

**Code Sample demonstrating the issue:**

```java
public class SomeClass {
      private Boolean bar;

      public void method(String s) {
            this.bar = new Boolean(s);       //violation for the BooleanInstantiation
            this.bar = Boolean.valueOf(s);  //use it instead of Boolean#new
      }
}
```

**Expected outcome:**

PMD should report a violation at line 5, but doesn't. This is a false-negative.

**Running PMD through:** *Maven*
